### DISCUSSION TITLE
Does agent recognize yaml configure file?

### DISCUSSION THREAD

**Discussion by changliu-wk** [open-telemetry/opentelemetry-java-instrumentation#6536] [Q&A] [answered]
Previous we use *.properties file to configure java agent, something like this: 
`otel.exporter.otlp.protocol = http/protobuf`, it works fine. 
But we have one project uses ymal file to do the configuration, they want something like this:
```
otel:
  propagators: tracecontext
  traces:
    exporter: otlp
```
I run my jar along with this:
`     -Dotel.javaagent.configuration-file=/otel/otel.properties.yaml \
      -javaagent:/otel/opentelemetry.jar \`
Docker has this print out:
`[otel.javaagent 2022-09-01 18:48:07:599 +0000] [main] INFO io.opentelemetry.javaagent.tooling.VersionLogger - opentelemetry-javaagent - version: 1.17.0`
But there is no output in the UI, I guess agent doesn't recognize yaml file, didn't config right?

**Comment by changliu-wk** [comment]
Seems the agent could get this kind of yaml? 
```
  - name: OTEL_PROPAGATORS
    value: tracecontext
```

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @changliu-wk ,
The javaagent does not support yaml config files.
There have been very general discussions about a standardized OpenTelemetry yaml config format that'd work across all language SDKs, but it's still in the idea phase (and it'll probably be a while before anything substantial is specced). For now, the options that we provide are system properties, environment variables and the properties config file.
